src/HPolyhedron.jl

```diff
@@ -499,6 +501,43 @@ function remove_redundant_constraints!(P::PT;
     return P
 end
 
+"""
+    linear_map(M::AbstractMatrix{N}, P::PT) where {N<:Real, PT<:HPoly{N}}
+
+Concrete linear map of a polyhedron in constraint representation.
+
+### Input
+
+- `M` -- matrix
+- `P` -- polyhedron in constraint representation
+
+### Output
+
+A polyhedron of the same type as the input (`PT`).
+
+### Algorithm
+
+If the matrix ``M`` is invertible (which we check with a sufficient condition),
+then ``y = M x`` implies ``x = \\text{inv}(M) y`` and we transform the
+constraint system ``A x ≤ b`` to ``A \\text{inv}(M) y ≤ b``.
+"""
+function linear_map(M::AbstractMatrix{N}, P::PT) where {N<:Real, PT<:HPoly{N}}
+    if !isinvertible_sufficient(M)
+        if P isa HPolyhedron
+            error("linear maps for polyhedra need to be invertible")
```

Comment:
this is a library requirement but i think that we should change and revisit this (take for example a bounded polyhedron); probably after https://github.com/JuliaReach/LazySets.jl/issues/998